UI/js/controllers/controllers.js

```diff
@@ -85,32 +85,64 @@ dashboardApp.controller('termsController', ['Courses', 'Terms', '$rootScope', '$
 
 dashboardApp.controller('scheduleController', ['Schedule', '$scope', '$rootScope', function(Schedule, $scope, $rootScope) {
   $scope.loadingSchedule = true;
-  Schedule.getSchedule('/todolms/' + $rootScope.user).then(function(data) {
+  $scope.schedule =[];
+  $scope.schedule_time_options = [{
+     name: 'Due last 7 days',
+     value: 'overdue'
+  }, {
+     name: 'Due today',
+     value: 'today'
+  }, {
+     name: 'Next 7 days',
+     value: 'week'
+  }];
+
+  $scope.scheduleErrors=[];
+
+  Schedule.getSchedule('/todolms/studenta/ctools').then(function(data) {
     $scope.loadingSchedule = false;
-    $scope.schedule = data.combinedSchedule;
-    $scope.scheduleStatus = data.status;
-    $scope.schedule_time_options = [{
-       name: 'Due last 7 days',
-       value: 'overdue'
-    }, {
-       name: 'Due today',
-       value: 'today'
-    }, {
-       name: 'Next 7 days',
-       value: 'week'
-    }];
-
-   $scope.$on('canvasCourses', function (event, canvasCourses) {
-      $.each($scope.schedule, function() {
-        if(this.contextLMS === 'canvas'){
-          var thisId = _.last(this.contextUrl.split('/'));
-          var thisContext = _.findWhere(canvasCourses, {id: thisId});
-          if(thisContext){
-            this.context = thisContext.title;
-          }          
+    if(data.status ===200){
+      $scope.schedule = data.data.Result.concat($scope.schedule);
+    } else {
+      $scope.scheduleErrors.push({'status':data.status, 'source':'ctools'});
+    }
+  });
+  Schedule.getSchedule('/todolms/studenta/ctoolspast').then(function(data) {
```

Comment:
If there is time later to reduce repetition that would be good.  This code is fine given the limited number of different urls.